launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,403 @@
+    private static final String EMAIL = "email"; 
+    private static final String ID = "id";
+
+    private static final String TEST_PODCAST_TITLE = "TEST_PODCAST_TITLE";
+    private static final String TEST_EMPTY_PODCAST_TITLE = "";
+    private static final String TEST_NULL_PODCAST_TITLE = null;
+    private static final String TEST_MP3_LINK = "TEST_MP3_LINK";
+    private static final String TEST_EMPTY_MP3_LINK = "";
+    private static final String TEST_NULL_MP3_LINK = null;
+    private static final long TEST_TIMESTAMP = System.currentTimeMillis();
+    private static final String TEST_ID = "123456";
+    private static final String TEST_ID_TWO = "789012";
+    private static final String TEST_XML_STRING;
+    private static final String TEST_PUBDATE = "2020/06/26 01:32:06";
+    private static final String TEST_EMAIL = "123@abc.com";
+    private static final String TEST_INCORRECT_EMAIL = "123@cde.com";
+    private static final String TEST_EMAIL_TWO = "456@abc.com";
+    private static final String EMPTY_STRING = "";
+    private static final String BASE_URL = "https://launchpod-step18-2020.appspot.com?id=";
+
+    @Before 
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        helper.setUp();
+    }
+
+    @After 
+    public void tearDown() {
+      helper.tearDown();
+    }
+
+    /**
+     * Creates a test user feed entity.
+    */
+    private Entity makeEntity(String title, String mp3Link, String xmlString) {
+        Entity userFeedEntity = new Entity(USER_FEED);
+        userFeedEntity.setProperty(PODCAST_TITLE, title);
+        userFeedEntity.setProperty(MP3_LINK, mp3Link);
+        userFeedEntity.setProperty(XML_STRING, xmlString);
+        return userFeedEntity;
+    }
+
+    /**
+     * Creates a test URL given the title of the podcast and the identifier key of its Datastore entity.
+    */
+    private String createUrl(String title, String id) {
+        String url = "<div><p>" + title + "URL: <a href=\"https://launchpod-step18-2020.appspot.com/?id=" + id + "\">" + id + "</a></p></div>";
+        return url;
+    }
+
+    /**
+     * Create RSS XML string given a user feed entity.
+     * @return xml String
+     * @throws IOException
+    */
+    private static String xmlString(RSS rssFeed) throws IOException {
+        XmlMapper xmlMapper = new XmlMapper();
+        String xmlString = xmlMapper.writeValueAsString(rssFeed);
+        return xmlString;
+    }
+
+    /**
+     * Asserts that doPost() takes in form inputs from client and successfully stores that information in a Datastore entity.
+     * @throws IOException
+    */
+    @Test
+    public void doPost_StoresCorrectFormInput() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        when(request.getParameter(PODCAST_TITLE)).thenReturn(TEST_PODCAST_TITLE);
+        when(request.getParameter(MP3_LINK)).thenReturn(TEST_MP3_LINK);
+
+        StringWriter stringWriter = new StringWriter();
```

Comment:
these 3 boilerplate lines can be put in the setUp method so you can remove it from all tests